rfcs/ImplicitInheritance.md

```diff
@@ -0,0 +1,236 @@
+
+With these changes, the above example may be written as follows.
+```graphql
+interface Vehicle {
+  id: ID!
+  make: VehicleMake
+  wheelCount: Int
+  acceleration: Float
+  topSpeed: Float
+  mass: Float
+}
+
+interface Automobile implements Vehicle {
+  engineCylinders: Int
+  engineSize: Float
+}
+
+interface AutomobileForPersonTransfer implements Automobile {
+  capacity: Int
+}
+
+type Bicycle implements Vehicle {
+  make: BicycleMake # We can still overwrite fields if needed
+  wheelCount: Int # Or repeat them 1:1, if preferred
+  capacity: Int
+  gearCount: Int
+}
+
+type Car implements AutomobileForPersonTransfer {
+  make: CarMake
+  trunkSize: Float
+}
+
+type Motorcycle implements AutomobileForPersonTransfer {
+  make: MotorcycleMake
+  topSpeed: SmallFloat
+  class: MotorcycleClass
+}
+
+type Truck implements Automobile {
+  make: TruckMake
+  maxLoad: Float
+}
+```
+
+## Advantages
+- Fields that changed or that are new are highlighted and stand out, which makes reading and understanding the schema easier.
+- No repetition in the definition of the fields.
+- The schema is easier to scale and maintain.
+  In particular, addition of new types implementing multiple interfaces is easier and doesn't involve copy & paste.
+- Changes to interface stay mostly local to that interface.
+  That is, addition or change of a field in an interface for the most part doesn't require changes in the implementing types.
+- Interfaces can be used as a kind of `fragment` on the schema-side by extracting common structures and making them easy to reuse. 
+- Shorter schema, which require less writing and are easier to skim/read.
+- Aligns with the handling of interfaces in other object-oriented languages such as TypeScript and with, e.g., the documentation of the Github GraphQL API which doesn't include inherited fields, see [for example](https://docs.github.com/en/graphql/reference/objects#issue).
+  Following established conventions leads to an easier knowledge transfer.
+  Moreover, it also aligns the way the schema is defined with other alternatives to SDL like [TypeGraphQL](https://typegraphql.com/).
+
+## Disadvantages
+- It is no longer possible to immediately see all fields that are provided by a type (without traversing the inheritance hierarchy), which makes reading the schema more challenging in some aspects.
+- If a type inherits multiple interfaces that declare a field with different but compatible types, the resulting type is not immediately clear.  
+
+It should be noted that these disadvantages can mostly be mitigated using tools or IDE integrations that can easily determine all fields on the fly.
+
+## Evaluation against [guiding principles](https://github.com/graphql/graphql-spec/blob/main/CONTRIBUTING.md#guiding-principles)
+- Backwards compatibility: ✅
+
+  Existing queries and schema are still valid.
```

Comment:
Yes, but some GraphQL frameworks/parsers most likely adhere to current "hard" rules about inheriting fields from interfaces so there may be problems when trying to use them with the "new lightweight" schema.